src/object-store/src/property.hpp

```diff
@@ -55,6 +55,7 @@ namespace realm {
 
         size_t table_column = -1;
         bool requires_index() const { return is_primary || is_indexed; }
+        bool is_indexable() const { return type == PropertyTypeInt || type == PropertyTypeBool || type == PropertyTypeString || type == PropertyTypeDate; }
```

Comment:
You should join the object store repo so you can give this feedback to the commiter of this code.